pep-0705.rst

```diff
@@ -0,0 +1,259 @@
+Finally, this allows bringing the benefits of ``TypedDict`` to other mapping types that are unrelated to ``dict``.
+
+Specification
+=============
+
+The ``TypedMapping`` type is a protocol behaving almost identically to ``TypedDict``, except:
+
+1. The runtime type of a TypedMapping object is not constrained to be a ``dict``
+2. no mutator methods (``__setitem__``, ``__delitem__``, ``update``, etc.) will be generated
+3. a class definition defines a ``TypedMapping`` protocol if and only if ``TypedMapping`` appears directly in its class bases
+4. subclasses can narrow field types, in the same manner as other protocols
+
+All current and future features of TypedDict are applicable to TypedMapping, including class-based and alternative syntax, totality, and ``Required`` and ``NotRequired`` from :pep:`655`.
+
+As with :pep:`589`, this PEP provides a sketch of how a type checker is expected to support type checking operations involving TypedMapping and TypedDict objects, but details are left to implementors. In particular, type compatibility should be based on structural compatibility.
+
+
+Multiple inheritance and TypedDict
+----------------------------------
+
+A type that inherits from a TypedMapping protocol and from TypedDict (either directly or indirectly):
+
+5. is the structural intersection of its parents, or invalid if no such intersection exists
+6. instances must be a dict subclass
+7. adds mutator methods only for fields it explicitly (re)declares
+
+For example::
+
+    class Movie(TypedMapping):
+        name: str
+        year: int | None
+
+    class MovieRecord(A, TypedDict):
+        year: int
+
+    movie: MovieRecord = { "name": "Blade Runner",
+                           "year": 1982 }
+     
+    movie["year"] = 1985  # Fine, mutator methods added in definition
+    movie["name"] = "Terminator"  # Type check error, "name" mutator not declared
+
+Inheriting, directly or indirectly, from both TypedDict and Protocol will continue to fail at runtime, and should continue to be rejected by type checkers.
+
+
+Multiple inheritance and Protocol
+---------------------------------
+
+* A type that inherits from a TypedMapping protocol and from a Protocol protocol must satisfy the protocols defined by both, but is not itself a protocol unless it inherits directly from TypedMapping or Protocol.
+* A type that inherits from a TypedMapping protocol and from Protocol itself is configured as a Protocol. Methods and properties may be defined; keys may not::
+
+    class A(Movie, Protocol):
+        # Declare a mutable property called 'year'
+        # This does not affect the dictionary key 'year'
+        year: str
+
+* A type that inherits from a Protocol protocol and from TypedMapping itself is configured as a TypedMapping. Keys may be defined; methods and properties may not::
+
+    class B(A, TypedMapping):
+        # Declare a key 'year'
+        # This does not affect the property 'year'
+        year: int
+
+
+Type Consistency Rules
```

Comment:
```suggestion
Type consistency rules
```

Either title case or sentence case is fine for non-top-level headings, so long as you pick one and are (relatively) consistent about it. My suggestion changes this one, but you could equally well change the previous two, if you preferred.